client/modules/workplace/actions/workplace_state.js

```diff
@@ -109,3 +112,25 @@ function resetData(LocalState) {
     LocalState.set('VIEW_OBJECT', viewObj);
   }
 }
+
+function setFieldsConstructorsType(fieldsList, LocalState) {
```

Comment:
lets use LocalState only inside `setSQLQuery` function. This `setFieldsConstructorsType` function should only return some result that we can use for setting to LocalState